core/tests/protractor_utils/CreatorDashboardPage.js

```diff
@@ -48,23 +61,69 @@ var CreatorDashboardPage = function() {
   };
 
   this.clickCreateActivityButton = function() {
-    createActivityButton.click();
+    browser.wait(until.elementToBeClickable(createActivityButton), 10000,
+      'Create Activity button takes too long to appear')
+      .then(function(isClickable) {
+        if (isClickable) {
+          createActivityButton.click();
+        }
+      });
   };
 
   this.clickCreateCollectionButton = function() {
-    createCollectionButton.click();
+    browser.wait(until.elementToBeClickable(createCollectionButton), 10000,
+      'Create Collection button takes too long to appear')
+      .then(function(isClickable) {
+        if (isClickable) {
+          createCollectionButton.click();
+        }
+      });
   };
 
   this.clickCreateExplorationButton = function() {
-    createExplorationButton.click();
+    browser.wait(until.elementToBeClickable(createExplorationButton), 5000,
+      'Create Exploration button takes too long to appear')
+      .then(function(isClickable) {
+        if (isClickable) {
+          createExplorationButton.click();
+        }
+      });
   };
 
   this.navigateToCollectionEditor = function() {
-    collectionCard.click();
+    browser.wait(until.elementToBeClickable(collectionCard), 5000,
+      'Collection Card tab takes too long to appear')
+      .then(function(isClickable) {
+        if (isClickable) {
+          collectionCard.click();
+        }
+      });
   };
 
   this.navigateToSubscriptionDashboard = function() {
-    subscriptionTab.click();
+    browser.wait(until.elementToBeClickable(subscriptionTab), 5000,
+      'Subscription Dashboard tab takes too long to appear')
+      .then(function(isClickable) {
+        if (isClickable) {
+          subscriptionTab.click();
+        }
+      });
+  };
+
+  this.editExploration = function(explorationTitle) {
+    _getExplorationElements(explorationTitle).then(function(elems) {
+      if (elems.length === 0) {
+        throw 'Could not find exploration tile with name ' + explorationTitle;
+      }
+      var explorationElement = elems[0].element(by.css('.mask-wrap'));
```

Comment:
Should use a CSS class that starts with .protractor-test-

Reply:
Done!